src/opentelemetry.ts

```diff
@@ -19,7 +19,7 @@ export const withOpenTelemetry = async <T>(opts: Options, f: () => Promise<T>):
     // This action should be run on workflow_run event,
     // the current environment does not reflect the target workflows.
     autoDetectResources: false,
-    resource: new Resource({
+    resource: resourceFromAttributes({
```

Comment:
https://github.com/open-telemetry/opentelemetry-js/blob/3c452f4b20f69383167712553a8c869d36b388dc/doc/upgrade-to-2.x.md